### ISSUE TITLE
Prefix/suffix/lowercase crashes on oneof

### ISSUE THREAD

**Issue by tsloughter** [tomas-abrahamsson/gpb#75]
If the options include a msg prefix, suffix or `msg_name_to_lower` the parser crashes in `prefix_suffix_fields` because it expects every list element to be a `gpb_field` but instead finds a `gpb_oneof`:

https://github.com/tomas-abrahamsson/gpb/blob/master/src/gpb_parse.yrl#L1093-L1101

**Comment by tsloughter** [comment]
Hm, the issue is a little more complicated than I thought.

If I have it simply skip the `oneof` in this function I still have an issue with both `oneof` and `map` if they use the types of other messages and a formatter like lowercase:

```
message A { ...}

message B { 
  oneof x {
    A f1 = 1;
  }

  map<string, A> y = 2;
}
```

Both of these will fail because it can't find `A` in a dict it looks it up in `is_msgsize_known_at_generationtime`.

Not sure yet where in the parser to properly format those types.

**Comment by tomas-abrahamsson** [comment]
For `oneof`, I think the following change would do it:
```
--- a/src/gpb_parse.yrl
+++ b/src/gpb_parse.yrl
@@ -1095,6 +1095,9 @@ prefix_suffix_fields(Prefix, Suffix, ToLower, Fields) ->
       fun(#?gpb_field{type={msg,MsgName}}=F) ->
               NewMsgName = prefix_suffix_name(Prefix, Suffix, ToLower, MsgName),
               F#?gpb_field{type={msg,NewMsgName}};
+         (#gpb_oneof{fields=Fs}=F) ->
+              Fs2 = prefix_suffix_fields(Prefix, Suffix, ToLower, Fs),
+              F#gpb_oneof{fields=Fs2};
          (#?gpb_field{}=F) ->
               F
       end,
```

**Comment by tomas-abrahamsson** [comment]
Will take a look at `map<_,_>`

**Comment by tomas-abrahamsson** [comment]
Good thing you thought about maps too. Perhaps something along these lines:
```
--- a/src/gpb_parse.yrl
+++ b/src/gpb_parse.yrl
@@ -1095,6 +1095,9 @@ prefix_suffix_fields(Prefix, Suffix, ToLower, Fields) ->
       fun(#?gpb_field{type={msg,MsgName}}=F) ->
               NewMsgName = prefix_suffix_name(Prefix, Suffix, ToLower, MsgName),
               F#?gpb_field{type={msg,NewMsgName}};
+         (#?gpb_field{type={map,KeyType,{msg,MsgName}}}=F) ->
+              NewMsgName = prefix_suffix_name(Prefix, Suffix, ToLower, MsgName),
+              F#?gpb_field{type={map,KeyType,{msg,NewMsgName}}};
          (#gpb_oneof{fields=Fs}=F) ->
               Fs2 = prefix_suffix_fields(Prefix, Suffix, ToLower, Fs),
               F#gpb_oneof{fields=Fs2};
```

**Comment by tsloughter** [comment]
Cool, yup, those did it, thanks!

**Comment by tomas-abrahamsson** [comment]
I merged your fix in #76 and just pushed it as 3.25.0. (Closing)
